src/Gui/Inventor/Draggers/Gizmo.cpp

```diff
@@ -0,0 +1,710 @@
+        },
+        this
+    );
+    dragger->addFinishCallback(
+        [] (void* data, SoDragger*) {
+            static_cast<LinearGizmo*>(data)->draggingFinished();
+        },
+        this
+    );
+    dragger->addMotionCallback(
+        [] (void* data, SoDragger*) {
+            static_cast<LinearGizmo*>(data)->draggingContinued();
+        },
+        this
+    );
+
+    dragger->labelVisible = false;
+
+    setDragLength(property->value().getValue());
+
+    dragger->instantiateBaseGeometry();
+
+    // change the dragger dimensions
+    auto arrow = SO_GET_PART(dragger, "arrow", SoArrowGeometry);
+    arrow->cylinderHeight = 3.5;
+    arrow->cylinderRadius = 0.2;
+
+    return draggerContainer;
+}
+
+void LinearGizmo::uninitDragger()
+{
+    dragger = nullptr;
+    draggerContainer = nullptr;
+}
+
+GizmoPlacement LinearGizmo::getDraggerPlacement()
+{
+    assert(draggerContainer && "Forgot to call GizmoContainer::initGizmos?");
+    return {draggerContainer->translation.getValue(), draggerContainer->getPointerDirection()};
+}
+
+void LinearGizmo::setDraggerPlacement(const SbVec3f& pos, const SbVec3f& dir)
+{
+    assert(draggerContainer && "Forgot to call GizmoContainer::initGizmos?");
+    draggerContainer->translation = pos;
+    draggerContainer->setPointerDirection(dir);
+}
+
+void LinearGizmo::reverseDir() {
+    auto dir = getDraggerContainer()->getPointerDirection();
+    getDraggerContainer()->setPointerDirection(dir * -1);
+}
+
+
+double LinearGizmo::getDragLength()
+{
+    double dragLength = dragger->translationIncrementCount.getValue()
+        * dragger->translationIncrement.getValue();
+
+    return (dragLength - addFactor) / multFactor;
+}
+
+void LinearGizmo::setDragLength(double dragLength)
+{
+    dragLength = dragLength * multFactor + addFactor;
+    dragger->translation = {0, static_cast<float>(dragLength), 0};
+}
+
+void LinearGizmo::setGeometryScale(float scale)
+{
+    dragger->geometryScale = SbVec3f(scale, scale, scale);
+}
+
+SoLinearDraggerContainer* LinearGizmo::getDraggerContainer()
+{
+    assert(draggerContainer && "Forgot to call GizmoContainer::initGizmos?");
+    return draggerContainer;
+}
+
+void LinearGizmo::setProperty(QuantitySpinBox* property)
+{
+    if (quantityChangedConnection) {
+        QuantitySpinBox::disconnect(quantityChangedConnection);
+    }
+
+    this->property = property;
+    quantityChangedConnection = QuantitySpinBox::connect(
+        property, qOverload<double>(&Gui::QuantitySpinBox::valueChanged),
+        [this] (double value) {
+            setDragLength(value);
+        }
+    );
+    quantityChangedConnection = QuantitySpinBox::connect(
+        property, &Gui::QuantitySpinBox::showFormulaDialog,
+        [this] (bool) {
+            if (this->property->hasExpression()) {
+                getDraggerContainer()->visible = false;
+            } else {
+                getDraggerContainer()->visible = true;
+            }
```

Comment:
```suggestion
            getDraggerContainer()->visible = this->property->hasExpression();
```

wouldn't that work? Also, I'm not sure if that is the correct way - if there is another thing that would should hide the dragger and I'll toggle the expression it will show up.